### ISSUE TITLE
VerifyError on Android since 1.3.7 (register has type Reference but expected Precise Reference)

### ISSUE THREAD

**Issue by ralexey** [Kotlin/kotlinx.coroutines#2049]
I saw the issue #2041 but looks like my issue is a different.
App works fine with version 1.3.6 and crashes on 1.3.7
With 1.3.7 app crashes on start.

Crashlog:
```
java.lang.VerifyError: Verifier rejected class com.volcast.data.VolcastRepositoryImpl: java.lang.Object com.volcast.data.VolcastRepositoryImpl.refresh(kotlin.coroutines.Continuation) failed to verify: java.lang.Object com.volcast.data.VolcastRepositoryImpl.refresh(kotlin.coroutines.Continuation): [0x12F] register v3 has type Reference: java.lang.Exception but expected Precise Reference: kotlin.jvm.internal.Ref$ObjectRef (declaration of 'com.volcast.data.VolcastRepositoryImpl' appears in /data/app/com.volcast.temp-6UrzwdYgEJRIKC9aa-RzVQ==/base.apk!classes2.dex)
        at com.volcast.di.DataModuleKt$dataModule$1$12.invoke(DataModule.kt:58)
        at com.volcast.di.DataModuleKt$dataModule$1$12.invoke(Unknown Source:4)
        at org.koin.core.instance.InstanceFactory.create(InstanceFactory.kt:50)
        at org.koin.core.instance.FactoryInstanceFactory.get(FactoryInstanceFactory.kt:36)
        at org.koin.core.registry.InstanceRegistry.resolveInstance$koin_core(InstanceRegistry.kt:87)
        at org.koin.core.scope.Scope.resolveInstance(Scope.kt:214)
        at org.koin.core.scope.Scope.get(Scope.kt:181)
        at com.volcast.di.ViewModelModuleKt$viewModelModule$1$7.invoke(ViewModelModule.kt:72)
        at com.volcast.di.ViewModelModuleKt$viewModelModule$1$7.invoke(Unknown Source:4)
...
```

code:
```
interface VolcastRepository {
    fun getVolcasts(): LiveData<ModelEvent<List<Volcast>>>
    suspend fun refresh()
}

class VolcastRepositoryImpl(
    private val volcastDao: VolcastDao,
    private val api: Api,
    private val mapperVolcastRemoteDb: MapperVolcastRemoteDb,
    mapperVolcastDbApp: MapperVolcastDbApp,
    private val dispatchersManager: DispatchersManager
) : VolcastRepository {
    private val _networkState = MutableLiveData<ModelEvent<Nothing>>()

    private val _volcast = RemoteLiveData(
        dbData = volcastDao.getVolcasts(),
        remoteState = _networkState,
        mapper = mapperVolcastDbApp
    )

    override fun getVolcasts(): LiveData<ModelEvent<List<Volcast>>> = _volcast

    override suspend fun refresh() {
        if (_networkState.value is ModelEvent.Start)
            return
        withContext(dispatchersManager.ui) {
            _networkState.value = ModelEvent.Start()
        }
        try {
            val volcasts = withContext(dispatchersManager.io) {
                api.getVolcasts()
            }
            withContext(dispatchersManager.default) {
                volcastDao.updateVolcasts(volcasts.volcasts?.map { mapperVolcastRemoteDb.map(it) })
            }
            withContext(dispatchersManager.ui) {
                _networkState.value = ModelEvent.Complete()
            }
        } catch (e: Exception) {
            withContext(dispatchersManager.ui) {
                _networkState.value = ModelEvent.Error(e.toAppException())
            }
        }
    }
}
```
If i remove `try` and `catch` - app starts ok but without crash handling.
Moving all code inside "try-catch" to the another function didn't help.

**Comment by qwwdfsad** [comment]
Could you please specify your Kotlin version?

As a workaround, please extract `withContext` (probably from `catch` block) to a separate function.

Unfortunately, if I'm watering down your example, the bug won't reproduce, so if you could simplify it a bit, it would be really great

**Comment by ralexey** [comment]
@qwwdfsad I moved all `withContext` from `try` and `catch` blocks to other functions and it helped me. App works as expected now.

**Comment by qwwdfsad** [comment]
Great!
Unfortunately, this is a compiler bug that will only be fixed in Kotlin 1.4 (https://youtrack.jetbrains.com/issue/KT-39113).

**Comment by ralexey** [comment]
Thank you for your help. Should i close the issue or you will do it yourself?

**Comment by elanpang** [comment]
revert coroutines lib to 1.3.6 works for me

**Comment by johngoren** [comment]
> Great!
> Unfortunately, this is a compiler bug that will only be fixed in Kotlin 1.4 (https://youtrack.jetbrains.com/issue/KT-39113), so we cannot fix it right now

Yeah, this has resolved for me by upgrading to Kotlin 1.4.10.

**Comment by walter-medina** [comment]
Sigo teniendo este problema, en local funciona, pero al pasar por jenkins se produce el error.
Android version 11.0
gradle:6.5

Cuando arriba dice que pasarse a la 1.3.6 se refieren a cambiar en el gradle del proyecto esto? ext.kotlin_version='1.3.6' ?

Fatal Exception: java.lang.VerifyError: Verifier rejected class co.com.wallet.colsubsidio.screens.register.view.fragment.RegisterScannerFragment: void co.com.wallet.colsubsidio.screens.register.view.fragment.RegisterScannerFragment.j() failed to verify: void co.com.wallet.colsubsidio.screens.register.view.fragment.RegisterScannerFragment.j(): [0x29C] register v3 has type Reference: java.lang.Object but expected Precise Reference: androidx.camera.core.CameraSelector (declaration of 'co.com.wallet.colsubsidio.screens.register.view.fragment.RegisterScannerFragment' appears in /data/app/com.colsubsidio.wallet.lab-T6E9SAzJi_SPLeutDgU9QQ==/base.apk)
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at androidx.fragment.app.Fragment.instantiate(Fragment.java:21)
       at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java)
       at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:9)
       at androidx.navigation.fragment.FragmentNavigator.instantiateFragment(FragmentNavigator.java:8)
       at androidx.navigation.fragment.FragmentNavigator.navigate(FragmentNavigator.java:55)
       at androidx.navigation.fragment.FragmentNavigator.navigate(FragmentNavigator.java:2)
       at androidx.navigation.NavController.navigate(NavController.java:38)
       at androidx.navigation.NavController.navigate(NavController.java:173)
       at androidx.navigation.NavController.navigate(NavController.java:1)
       at androidx.navigation.NavController.navigate(NavController.java:1)
       at androidx.navigation.NavController.navigate(NavController.java:1)
       at co.com.wallet.colsubsidio.screens.register.view.fragment.IdentificationPageViewFragment.goToScanner(IdentificationPageViewFragment.java:1)
       at androidx.fragment.app.FragmentActivity.onRequestPermissionsResult(FragmentActivity.java:68)
       at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:8408)
       at android.app.Activity.dispatchActivityResult(Activity.java:8253)
       at android.app.ActivityThread.deliverResults(ActivityThread.java:4970)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:5018)
       at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2073)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:225)
       at android.app.ActivityThread.main(ActivityThread.java:7563)
